src/NHibernate/Dialect/FirebirdDialect.cs

```diff
@@ -520,6 +520,12 @@ private void RegisterTrigonometricFunctions()
 			RegisterFunction("tanh", new StandardSQLFunction("tanh", NHibernateUtil.Double));
 		}
 
+		// As of Firebird 2.5 documentation, limit is 30/31 (not all source are concordant), with some
+		// cases supporting more but considered as bugs and no more tolerated in v3.
+		// It seems it may be extended to 63 for Firebird v4.
+		/// <inheritdoc />
+		public override int MaxAliasLength => 30;
```

Comment:
It seems we still have a bug. A test has generated the alias `requestcompletionstatus3_8691_0_` for a query, that is 32 characters long.

Reply:
It's because the identifier has 2 "unique" parts for some reasons.

Reply:
It's in "Column.cs", line 173.

Reply:
Or not. I'm confused.

Reply:
I am going to have a look. I am currently adjusting lengths and sources for those lengths.

Reply:
It is from `SelectFragment.cs` suffix field. Which is a "collectionSuffix" which is a table number + `_`.

Reply:
Run one by one, the tests succeed. It seems there is some coupling between tests.

Reply:
The coupling is very likely coming from the thread static `tableCounter` in `Table` class. It seems it endlessly grows as we build/discard session factories.
Either this field should be reset when starting to build a session factory (if it has no more usages after build) or it should be bound to the session factory instead of the thread on which it was built.

The `SelectFragment` suffix is an additional suffix which is meant (at least in the cases I have checked) to differentiate columns of the same table in case it is auto-joined. It does not detect auto-join but simply always add the suffix according to query joins count.

It then means the `Column.GetAlias` should account for that potential additional suffix by not exhausting the full maximum length. At least 3 characters should be left in my opinion, for supporting more than 9 joins (up to 99) and the additional `_` of the suffix.

If we want want to support more than 99 joins (up to 999), we should keep 4 characters left, but I am not used to have such monstrous queries with NHibernate.